### ISSUE TITLE
LoadBalancer using all members even though 2/3 are unhealthy

### ISSUE THREAD

**Issue by bhechinger** [containous/traefik#6314]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I create a load balancer configuration for Vault.

### What did you expect to see?

I expected to see Traefik not route to the unhealthy service members.

### What did you see instead?

I see Traefik routing to the unhealthy service members.

### Output of `traefik version`: (_What version of Traefik are you using?_)

Built from the 2.1.3 branch:
```
$ traefik version
Version:      c796cd2250ecdf9dd421f78b6d1cf621076b32ef
Codename:     cheddar
Go version:   go1.13.4
Built:        2020-01-29_09:18:44PM
OS/Arch:      illumos/amd64

```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```yaml
http:
  routers:
    shared-vault-http:
      entryPoints:
        - web
      rule: "Host(`vault.example.net`)"
      service: shared-vault
      middlewares:
        - redirect

    shared-vault-https:
      entryPoints:
        - websecure
      rule: "Host(`vault.example.net`)"
      service: shared-vault
      tls:
        certResolver: cloudflare

  services:
    shared-vault:
      loadBalancer:
        healthCheck:
          path: /v1/sys/health
          port: 8200
          scheme: https
          interval: "10s"
          timeout: "3s"
        servers:
          - url: https://vault-01.example.net:8200
          - url: https://vault-02.example.net:8200
          - url: https://vault-03.example.net:8200

  middlewares:
    redirect:
      redirectScheme:
        scheme: https
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

I see in the Traefik logs where vault-01/03 are failing healthchecks and it claims it's removing them from the pool. In the UI I see 01/03 marked as unhealthy yet I'm still having requests sent to them.

```
$ curl -i https://vault.example.net/v1/sys/health
HTTP/2 503 
cache-control: no-store
content-type: application/json
date: Wed, 12 Feb 2020 21:51:46 GMT
content-length: 198
$ curl -i https://vault.example.net/v1/sys/health
HTTP/2 200 
cache-control: no-store
content-type: application/json
date: Wed, 12 Feb 2020 21:51:46 GMT
content-length: 294
$ curl -i https://vault.exmaple.net/v1/sys/health
HTTP/2 429 
cache-control: no-store
content-type: application/json
date: Wed, 12 Feb 2020 21:51:47 GMT
content-length: 291
```

**Comment by bhechinger** [comment]
```time="2020-02-13T11:28:48-05:00" level=warning msg="Health check still failing. Backend: \"shared-vault@file\" URL: \"https://vault-01.4amlunch.net:8200\" Reason: received error status code: 503"
time="2020-02-13T11:28:48-05:00" level=warning msg="Health check still failing. Backend: \"shared-vault@file\" URL: \"https://vault-03.4amlunch.net:8200\" Reason: received error status code: 429"
time="2020-02-13T11:28:48-05:00" level=warning msg="Health check still failing. Backend: \"shared-vault@file\" URL: \"https://vault-01.4amlunch.net:8200\" Reason: received error status code: 503"
time="2020-02-13T11:28:48-05:00" level=warning msg="Health check still failing. Backend: \"shared-vault@file\" URL: \"https://vault-03.4amlunch.net:8200\" Reason: received error status code: 429"
time="2020-02-13T11:28:48-05:00" level=warning msg="Health check still failing. Backend: \"shared-vault@file\" URL: \"https://vault-01.4amlunch.net:8200\" Reason: received error status code: 503"
time="2020-02-13T11:28:48-05:00" level=warning msg="Health check still failing. Backend: \"shared-vault@file\" URL: \"https://vault-03.4amlunch.net:8200\" Reason: received error status code: 429"```

**Comment by AndyBan** [comment]
I have a similar setup to the above with file backend and am getting the same issue.  Tested on 2.1.4 and 2.0.5 which was before some health check changes and both show the issue.

I did notice if I bring up all servers then take one offline the health check works as expected however if I start Traefik with a server down then it will be kept in rotation even though its marked as down.

**Comment by ldez** [comment]
Fixed by #6372
